src/proxy/http/metrics/service.rs

```diff
@@ -363,39 +357,58 @@ where
     C: ClassifyEos<Error = h2::Error>,
     C::Class: Hash + Eq,
 {
-    fn record_class(&mut self, class: Option<C::Class>) {
+    fn record_latency(&mut self) {
         let now = clock::now();
-        let lock = match self.metrics.take() {
+
+        let lock = match self.metrics.as_mut() {
             Some(lock) => lock,
             None => return,
         };
         let mut metrics = match lock.lock() {
             Ok(m) => m,
             Err(_) => return,
         };
+
         (*metrics).last_update = now;
 
-        let first_byte_at = self.first_byte_at.unwrap_or_else(|| now);
-        let class_metrics = match class {
-            Some(c) => metrics
-                .by_class
-                .entry(c)
-                .or_insert_with(|| ClassMetrics::default()),
-            None => &mut metrics.unclassified,
+        let status_metrics = metrics
+            .by_status
+            .entry(self.status)
+            .or_insert_with(|| StatusMetrics::default());
```

Comment:
Nit: [`or_default`](https://docs.rs/indexmap/1.0.2/indexmap/map/enum.Entry.html#method.or_default)?

Reply:
still doesn't exist on indexmap

Reply:
the link above is to the `IndexMap` docs...maybe we're not on a new enough version?